ui/app/pages/send/send-content/add-recipient/add-recipient.component.js

```diff
@@ -0,0 +1,236 @@
+import React, { Component } from 'react'
+import PropTypes from 'prop-types'
+import Fuse from 'fuse.js'
+import Identicon from '../../../../components/ui/identicon'
+import {isValidAddress} from '../../../../helpers/utils/util'
+import Dialog from '../../../../components/ui/dialog'
+import ContactList from '../../../../components/app/contact-list'
+import RecipientGroup from '../../../../components/app/contact-list/recipient-group/recipient-group.component'
+import {ellipsify} from '../../send.utils'
+
+export default class AddRecipient extends Component {
+
+  static propTypes = {
+    className: PropTypes.string,
+    query: PropTypes.string,
+    ownedAccounts: PropTypes.array,
+    addressBook: PropTypes.array,
+    updateGas: PropTypes.func,
+    updateSendTo: PropTypes.func,
+    ensResolution: PropTypes.string,
+    toError: PropTypes.string,
+    toWarning: PropTypes.string,
+    ensResolutionError: PropTypes.string,
+    selectedToken: PropTypes.object,
+    hasHexData: PropTypes.bool,
+    tokens: PropTypes.array,
+  }
+
+  static contextTypes = {
+    t: PropTypes.func,
+    metricsEvent: PropTypes.func,
+  }
+
+  state = {
+    isShowingTransfer: false,
+    isShowingAllRecent: false,
+  }
+
+  selectRecipient = (to, nickname = '') => {
+    const { updateSendTo, updateGas } = this.props
+
+    updateSendTo(to, nickname)
+    updateGas({ to })
+  }
+
+  searchForContacts = () => {
+    const { addressBook, query } = this.props
+    let contacts = addressBook.filter(({ name }) => !!name)
+
+    if (query) {
+      if (!this.contactFuse) {
+        this.contactFuse = new Fuse(contacts, {
```

Comment:
This might be better off attached to the class itself as a property so that we don't need to manage the state ourselves—as a property it would be created once when the class is constructed.